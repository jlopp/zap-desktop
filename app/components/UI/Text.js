import React from 'react'
import PropTypes from 'prop-types'
import { Text as BaseText } from 'rebass'

/**
 * @render react
 * @name Text
 * @example
 * <Text>Some text</Text>
 */
class Text extends React.PureComponent {
  static displayName = 'Text'

  static propTypes = {
    children: PropTypes.node
  }

  render() {
    const { children } = this.props
    return (
      <BaseText fontSize="m" {...this.props}>
        {children}
      </BaseText>
    )
  }
}

export default Text
