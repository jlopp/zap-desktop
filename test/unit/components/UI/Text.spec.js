import React from 'react'
import Text from 'components/UI/Text'
import renderer from 'react-test-renderer'

describe('component.UI.Text', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Text />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
