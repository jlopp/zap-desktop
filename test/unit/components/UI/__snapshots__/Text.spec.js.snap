// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Text should render correctly 1`] = `
.c0 {
  font-size: m;
}

<div
  className="c0"
  fontSize="m"
/>
`;
